clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -3430,6 +3430,7 @@ Sema::InstantiateClass(SourceLocation PointOfInstantiation,
     llvm::raw_string_ostream OS(Name);
     Instantiation->getNameForDiagnostic(OS, getPrintingPolicy(),
                                         /*Qualified=*/true);
+    OS << ", file:" << SourceMgr.getFilename(Instantiation->getLocation());
```

Comment:
I think we might need to get the spelling location / expansion location first.
Could you check for instantiations coming from various complicaged source locations and make sure we're doing the right thing??
- from macro expansions,
- from macro arguments,
- from concatenation operator `##`?

Reply:
Done.